flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/DataStreamCsvITCase.java

```diff
@@ -0,0 +1,363 @@
+                    + "city='"
+                    + city
+                    + '\''
+                    + ", lat="
+                    + lat
+                    + ", lng="
+                    + lng
+                    + ", country='"
+                    + country
+                    + '\''
+                    + ", iso2='"
+                    + iso2
+                    + '\''
+                    + ", adminName='"
+                    + adminName
+                    + '\''
+                    + ", capital='"
+                    + capital
+                    + '\''
+                    + ", population="
+                    + population
+                    + '}';
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            CityPojo that = (CityPojo) o;
+            return population == that.population
+                    && Objects.equals(city, that.city)
+                    && Objects.equals(lat, that.lat)
+                    && Objects.equals(lng, that.lng)
+                    && Objects.equals(country, that.country)
+                    && Objects.equals(iso2, that.iso2)
+                    && Objects.equals(adminName, that.adminName)
+                    && Objects.equals(capital, that.capital);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(city, lat, lng, country, iso2, adminName, capital, population);
+        }
+    }
+
+    private static <T> List<T> initializeSourceAndReadData(File testDir, StreamFormat<T> csvFormat)
+            throws Exception {
+        final FileSource<T> source =
+                FileSource.forRecordStreamFormat(csvFormat, Path.fromLocalFile(testDir)).build();
+
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(PARALLELISM);
+        env.setRestartStrategy(RestartStrategies.fixedDelayRestart(1, 0));
+
+        final DataStream<T> stream =
+                env.fromSource(source, WatermarkStrategy.noWatermarks(), "file-source");
+
+        return getResultsFromStream(stream);
+    }
+
+    @NotNull
+    private static <T> List<T> getResultsFromStream(DataStream<T> stream) throws Exception {
+        final ClientAndIterator<T> client =
+                DataStreamUtils.collectWithClient(stream, "Bounded Results Fetch");
+
+        final List<T> result = new ArrayList<>();
+        while (client.iterator.hasNext()) {
+            T next = client.iterator.next();
+            result.add(next);
+        }
+        return result;
+    }
+
+    // ------------------------------------------------------------------------
+    //  test data
+    // ------------------------------------------------------------------------
+
+    private static final String[] CSV_LINES =
```

Comment:
Can you move all static variables to the beginning of the class? The variable declaration should be always at the top of the file that other developers do not have to search them first. You can still generate the test data in a method and put it here.

Reply:
Moved static. I don't quite understand what you have in mind with regarding the method.

Reply:
No worries I meant something like

```java
private static final String[] CSV_LINES = getCsvLines();
```
to prevent having a lot of lines for the initialization just for next time :)